docs/misc/config.md

```diff
@@ -0,0 +1,131 @@
+    * A class representing the data type of the config value
+
+!!! note
+    `DoubleValue`s, `IntValue`s, and `LongValue`s are range values which specify the class as `Double`, `Integer`, and `Long` respectively.
+
+* **Whitelisted Values**
+  * Description: Value must be in supplied collection
+  * Class Type: `T`
+  * Method Name: `#defineInList`
+  * Additional Components:
+    * A collection of the allowed values the configuration can be
+
+* **List Values**
+  * Description: Value is a list of entries
+  * Class Type: `List<T>`
+  * Method Name: `#defineList`, `#defineListAllowEmpty` if list can be empty
+  * Additional Components:
+    * A validator to make sure a deserialized element from the list is valid
+
+* **Enum Values**
+  * Description: An enum value in the supplied collection
+  * Class Type: `Enum<T>`
+  * Method Name: `#defineEnum`
+  * Additional Components:
+    * A getter to convert a string or integer into an enum
+    * A collection of the allowed values the configuration can be
+
+* **Boolean Values**
+  * Description: A `boolean` value
+  * Class Type: `Boolean`
+  * Method Name: `#define`
+
+Registering a Configuration
+---------------------------
+
+Once a `ForgeConfigSpec` has been built, it must be registered to allow Forge to load, track, and sync the configuration settings as required. Configurations should be registered in the mod constructor via `ModLoadingContext#registerConfig`. A configuration can be registered with a given type representing the side the config belongs to, the `ForgeConfigSpec`, and optionally a specific file name for the configuration.
+
+Type   | Loaded | Synced to Client | Stored                         | Default Suffix
+:---:  | :---:  | :---:            | :---:                          | :---
+CLIENT | Client | ❌               | `.minecraft/config`            | `-client`
+COMMON | Both   | ❌               | `.minecraft/config`            | `-common`
+SERVER | Server | ✔️               | `<server_folder>/serverconfig` | `-server`
+
+!!! tip
+    Forge documents the [config types][type] within their codebase.
+
+```java
+// In the mod constructor with a ForgeConfigSpec CONFIG
+ModLoadingContext.get().registerConfig(Type.COMMON, CONFIG);
+```
+
+Listening for Configuration Loading/Reloading
+---------------------------------------------
+
+Operations that occur whenever a config is loaded or reloaded can be done using the `ModConfigEvent$Loading` and `ModConfigEvent$Reloading` events. The events must be [registered][events] to the mod event bus.
+
+!!! warning
+    These events are called for all configurations for the mod; the `ModConfig` object provided should be used to denote which configuration is being loaded or reloaded.
+
+[toml]: https://toml.io/
+[nightconfig]: https://github.com/TheElectronWill/night-config
+[type]: https://github.com/MinecraftForge/MinecraftForge/blob/1.18.x/fmlcore/src/main/java/net/minecraftforge/fml/config/ModConfig.java#L108-L136
```

Comment:
It may be better to make this a permalink to the line (commit rather than branch), for future-proofing.